conf/route_control.py

```diff
@@ -1,530 +1,851 @@
-    else:
-        raise Exception('BESS connection failure.')
-
-    print('Done.')
-
-
-def reconfigure(number, frame):
-    print('Received: {} Reloading routes'.format(number))
-    # clear arpcache
-    for ip in list(arpcache):
-        item = arpcache.get(ip)
-        del item
-    arpcache.clear()
-    for ip in list(neighborcache):
-        item = neighborcache.get(ip)
-        del item
-    neighborcache.clear()
-    for modname in list(modgatecnt):
-        item = modgatecnt.get(modname)
-        del item
-    modgatecnt.clear()
-    bootstrap_routes()
-    signal.pause()
+    Does not wait for a response. Expected to have the side
+    effect of populating the arp table entry for neighbor_ip.
+    """
+    logger.info("Sending ping to %s", neighbor_ip)
+    send(IP(dst=neighbor_ip) / ICMP())
 
 
-def cleanup(number, frame):
-    ipdb.unregister_callback(event_callback)
-    print('Received: {} Exiting'.format(number))
-    sys.exit()
+def fetch_mac(ipdb: IPDB, target_ip: str) -> Optional[str]:
+    """Fetches the MAC address of the target IP from the ARP table using IPDB.
 
+    Args:
+        ipdb (IPDB): The IPDB object.
+        target_ip (str): The target IP address.
 
-def main():
-    global arpcache, neighborcache, modgatecnt, ipdb, event_callback, bess, ipr
-    # for holding unresolved ARP queries
-    arpcache = {}
-    # for holding list of registered neighbors
-    neighborcache = {}
-    # for holding gate count per route module
-    modgatecnt = {}
-    # for interacting with kernel
-    ipdb = IPDB()
-    ipr = IPRoute()
-    # for bess client
-    bess = BESS()
+    Returns:
+        Optional[str]: The MAC address of the target IP.
+    """
+    neighbors = ipdb.nl.get_neighbours(dst=target_ip)
+    for neighbor in neighbors:
+        attrs = dict(neighbor['attrs'])
+        if attrs.get(KEY_NETWORK_LAYER_DEST_ADDR, '') == target_ip:
+            logger.info(
+                "Mac address found for %s, Mac: %s",
+                target_ip, attrs.get(KEY_LINK_LAYER_ADDRESS, ''),
+            )
+            return attrs.get(KEY_LINK_LAYER_ADDRESS, '')
+    logger.info("Mac address not found for %s", target_ip)
+    return None
 
-    # connect to bessd
-    connect_bessd()
 
-    # program current routes
-    bootstrap_routes()
+def mac_to_int(mac: str) -> int:
+    """Converts a MAC address to an integer."""
+    try:
+        return int(mac.replace(":", ""), 16)
+    except ValueError:
+        raise ValueError("Invalid MAC address: %s", mac)
 
-    # listen for netlink events
-    print('Registering netlink event listener callback...'),
-    event_callback = ipdb.register_callback(netlink_event_listener)
-    print('Done.')
 
-    signal.signal(signal.SIGHUP, reconfigure)
-    signal.signal(signal.SIGINT, cleanup)
-    signal.signal(signal.SIGTERM, cleanup)
-    signal.pause()
+def mac_to_hex(mac: str) -> str:
+    """Converts a MAC address to a hexadecimal string."""
+    return '{:X}'.format(mac_to_int(mac))
```

Comment:
If the MAC address begins with `0(s)`, this conversion (hex->int->hex) will return a "shorter" (than 6 octets) MAC address. Would it make sense to add a check or add logic to handle that case in here?

Reply:
Good catch, fixed.